client/src/app/metadata-schema-form/metadata-form/metadata-form.component.spec.ts

```diff
@@ -0,0 +1,105 @@
+import {async, ComponentFixture, TestBed} from '@angular/core/testing';
+
+import SpyObj = jasmine.SpyObj;
+import {MetadataFormComponent} from './metadata-form.component';
+import {JsonSchema} from '../models/json-schema';
+import {MetadataFormService} from '../metadata-form.service';
+import {MetadataForm} from '../models/metadata-form';
+import {MetadataFormConfig} from '../models/metadata-form-config';
+import {FormControl, FormGroup, Validators} from '@angular/forms';
+import {MetadataFormTab} from '../models/metadata-form-layout';
+
+function getSchema() {
+  return {
+    'name': 'project'
+  } as JsonSchema;
+}
+
+function getData() {
+  return {
+    content: null,
+    admin: {
+      key: 'value'
+    }
+  };
+}
+
+function createTab(title) {
+  return {
+    title: title,
+    key: '1.' + title,
+    items: [
+      '1.' + title
+    ]
+  } as MetadataFormTab;
+}
+
+function createMetadataFormConfig(tabs: MetadataFormTab[]) {
+  return {
+    viewMode: true,
+    removeEmptyFields: true,
+    layout: {
+      tabs: tabs
+    }
+  } as MetadataFormConfig;
+}
+
+function createFormGroup() {
+  return new FormGroup({
+    x: new FormControl('', Validators.required),
+    y: new FormControl('', Validators.required),
+    z: new FormControl('', Validators.required),
+  });
+}
+
+function createMetadataForm(tabs: MetadataFormTab[]): MetadataForm {
+  return {
+    key: 'test key',
+    jsonSchema: getSchema(),
+    config: createMetadataFormConfig(tabs),
+    metadataRegistry: jasmine.createSpyObj('MetadataRegistry', ['buildMetadataRegistry']),
+    formGroup: createFormGroup()
+  } as MetadataForm;
+}
+
+describe('MetadataFormComponent', () => {
+  let component: MetadataFormComponent;
+  let fixture: ComponentFixture<MetadataFormComponent>;
+
+  let metadataFormSvc: SpyObj<MetadataFormService>;
+
+  const contentTab = createTab('content');
+  const adminTab = createTab('admin');
+  const tabs = [contentTab, adminTab];
+
+  beforeEach(async(() => {
+    metadataFormSvc = jasmine.createSpyObj('MetadataFormService', ['createForm']);
+    metadataFormSvc.createForm.and.returnValue(createMetadataForm(tabs));
+    TestBed.configureTestingModule({
+      providers: [
+        { provide: MetadataFormService, useValue: metadataFormSvc},
+      ],
+      declarations: [MetadataFormComponent]
+    })
+      .compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(MetadataFormComponent);
+    component = fixture.componentInstance;
+    component.schema = getSchema();
+    component.data = getData();
+    component.config = createMetadataFormConfig(tabs);
+    component.selectedTabKey = 'content';
+    component.visibleTabs = tabs;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('tabIsVisible should pass with null content', () => {
+    component.tabIsVisible(contentTab);
```

Comment:
don't you need to make the assertion here?

Reply:
I updated the test case as the previous one did not really reflected the condition of the bug.
The bug was a null error that occurred when the project content was null and the UI wanted to render it (iterate over the content). In that case this test would fail.
But I can add an expectation. Give me a sec.

Reply:
I added it.